In [1]:
# import 'Pandas' 
import pandas as pd 

# import 'Numpy' 
import numpy as np

# import subpackage of Matplotlib
import matplotlib.pyplot as plt

# import 'Seaborn' 
import seaborn as sns

# to suppress warnings 
from warnings import filterwarnings
filterwarnings('ignore')

# display all columns of the dataframe
pd.options.display.max_columns = None

# display all rows of the dataframe
pd.options.display.max_rows = None
 
# to display the float values upto 3 decimal places     
pd.options.display.float_format = '{:.3f}'.format

**Import Dataset**

In [2]:
google=pd.read_csv("googleplaystore.csv")

**1. Display Top 5 Rows of The Dataset**

In [3]:
google.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN   4.100   
1                                Coloring book moana  ART_AND_DESIGN   3.900   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN   4.700   
3                              Sketch - Draw & Paint  ART_AND_DESIGN   4.500   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN   4.300   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

**2. Check the Last 3 Rows of The Dataset**

In [4]:
google.tail(3)

App             Category  \
10838                         Parkinson Exercices FR              MEDICAL   
10839                  The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840  iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
10838     NaN       3                9.5M       1,000+  Free     0   
10839   4.500     114  Varies with device       1,000+  Free     0   
10840   4.500  398307                 19M  10,000,000+  Free     0   

      Content Rating             Genres      Last Updated         Current Ver  \
10838       Everyone            Medical  January 20, 2017                   1   
10839     Mature 17+  Books & Reference  January 19, 2015  Varies with device   
10840       Everyone          Lifestyle     July 25, 2018  Varies with device   

              Android Ver  
10838          2.2 and up  
10839  Varies with device  
10840  Varies with device

**3. Find Shape of Our Dataset (Number of Rows & Number of Columns)**

In [5]:
google.shape

(10841, 13)

- The shape of 10841 rows, 13 columns.

**4. Get Information About Our Dataset Like Total Number Rows, Total Number of Columns, Datatypes of Each Column And Memory Requirement**

In [6]:
google.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


- There are 1 numerical column and 12 categorical column.

**5. Get Overall Statistics About The Dataframe**

In [7]:
google.describe(include="all").T

count unique                 top   freq  mean   std   min  \
App               10841   9660              ROBLOX      9   NaN   NaN   NaN   
Category          10841     34              FAMILY   1972   NaN   NaN   NaN   
Rating         9367.000    NaN                 NaN    NaN 4.193 0.537 1.000   
Reviews           10841   6002                   0    596   NaN   NaN   NaN   
Size              10841    462  Varies with device   1695   NaN   NaN   NaN   
Installs          10841     22          1,000,000+   1579   NaN   NaN   NaN   
Type              10840      3                Free  10039   NaN   NaN   NaN   
Price             10841     93                   0  10040   NaN   NaN   NaN   
Content Rating    10840      6            Everyone   8714   NaN   NaN   NaN   
Genres            10841    120               Tools    842   NaN   NaN   NaN   
Last Updated      10841   1378      August 3, 2018    326   NaN   NaN   NaN   
Current Ver       10833   2784  Varies with device   1459   NaN   NaN   NaN   
Android Ver       10838     33          4.1 and up   2451   NaN   NaN   NaN   

                 25%   50%   75%    max  
App              NaN   NaN   NaN    NaN  
Category         NaN   NaN   NaN    NaN  
Rating         4.000 4.300 4.500 19.000  
Reviews          NaN   NaN   NaN    NaN  
Size             NaN   NaN   NaN    NaN  
Installs         NaN   NaN   NaN    NaN  
Type             NaN   NaN   NaN    NaN  
Price            NaN   NaN   NaN    NaN  
Content Rating   NaN   NaN   NaN    NaN  
Genres           NaN   NaN   NaN    NaN  
Last Updated     NaN   NaN   NaN    NaN  
Current Ver      NaN   NaN   NaN    NaN  
Android Ver      NaN   NaN   NaN    NaN

**6. Total Number of App Titles Contain Astrology**

In [8]:
google.head(2)

App        Category  Rating  \
0  Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN   4.100   
1                             Coloring book moana  ART_AND_DESIGN   3.900   

  Reviews Size  Installs  Type Price Content Rating  \
0     159  19M   10,000+  Free     0       Everyone   
1     967  14M  500,000+  Free     0       Everyone   

                      Genres      Last Updated Current Ver   Android Ver  
0               Art & Design   January 7, 2018       1.0.0  4.0.3 and up  
1  Art & Design;Pretend Play  January 15, 2018       2.0.0  4.0.3 and up

In [9]:
count=0
for i in google.App:
    #print(i.split())
    if "Astrology" in i.split():
        count+=1
        print(i)
        
print(count)

Horoscopes – Daily Zodiac Horoscope and Astrology
သိင်္ Astrology - Min Thein Kha BayDin
iHoroscope - 2018 Daily Horoscope & Astrology
3


**7. Find Average App Rating**

In [10]:
print("Average Ratings of the Apps: ", round(google.Rating.mean(),3))

Average Ratings of the Apps:  4.193


**8. Find Total Number of Unique Category**

In [11]:
print("Number of  unique Category : " , google.Category.nunique())

Number of  unique Category :  34


**9. Which Category Getting The Highest Average Rating?**

In [12]:
google.Rating.max()

19.0

In [13]:
google["Category"][google["Rating"]==google["Rating"].max()]

10472    1.9
Name: Category, dtype: object

- 1.9 has maximum rating.

**10.Find Total Number of App having 5 Star Rating**

In [14]:
print( "Total Number of App having 5 Star Rating : ",len(google[google.Rating==5]))

Total Number of App having 5 Star Rating :  274


**11.Find Average Value of Reviews**

In [15]:
google.Reviews.dtype

dtype('O')

In [16]:
for i in google.Reviews:
    if "M" in i:
        print(i.split('M'))
        google.Reviews.replace(i , float(i.split('M')[0])*100000, inplace=True)

['3.0', '']


In [17]:
google.Reviews.dtype

dtype('O')

In [18]:
google.Reviews=google.Reviews.astype("float64")

In [19]:
print("Average value of Reviews: ", google.Reviews.mean())

Average value of Reviews:  444139.5990222304


**12.Find Total Number of Free and Paid Apps**

In [20]:
google.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN   4.100   
1                                Coloring book moana  ART_AND_DESIGN   3.900   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN   4.700   
3                              Sketch - Draw & Paint  ART_AND_DESIGN   4.500   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN   4.300   

     Reviews  Size     Installs  Type Price Content Rating  \
0    159.000   19M      10,000+  Free     0       Everyone   
1    967.000   14M     500,000+  Free     0       Everyone   
2  87510.000  8.7M   5,000,000+  Free     0       Everyone   
3 215644.000   25M  50,000,000+  Free     0           Teen   
4    967.000  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [21]:
google.Type.value_counts()

Free    10039
Paid      800
0           1
Name: Type, dtype: int64

In [22]:
google.Type.replace("0", "Free", inplace=True)

In [23]:
google.Type.value_counts()

Free    10040
Paid      800
Name: Type, dtype: int64

**13. Which App Has Maximum Reviews?**

In [24]:
google.Reviews.max()

78158306.0

In [25]:
google["App"][google.Reviews==google.Reviews.max()]

2544    Facebook
Name: App, dtype: object

- Facebook is the app with maximum reviews.

**14. Display Top 5 Apps Having Highest Reviews**

In [26]:
google.sort_values(by="Reviews", ascending=False).head()

App       Category  Rating      Reviews  \
2544            Facebook         SOCIAL   4.100 78158306.000   
3943            Facebook         SOCIAL   4.100 78128208.000   
336   WhatsApp Messenger  COMMUNICATION   4.400 69119316.000   
381   WhatsApp Messenger  COMMUNICATION   4.400 69119316.000   
3904  WhatsApp Messenger  COMMUNICATION   4.400 69109672.000   

                    Size        Installs  Type Price Content Rating  \
2544  Varies with device  1,000,000,000+  Free     0           Teen   
3943  Varies with device  1,000,000,000+  Free     0           Teen   
336   Varies with device  1,000,000,000+  Free     0       Everyone   
381   Varies with device  1,000,000,000+  Free     0       Everyone   
3904  Varies with device  1,000,000,000+  Free     0       Everyone   

             Genres    Last Updated         Current Ver         Android Ver  
2544         Social  August 3, 2018  Varies with device  Varies with device  
3943         Social  August 3, 2018  Varies with device  Varies with device  
336   Communication  August 3, 2018  Varies with device  Varies with device  
381   Communication  August 3, 2018  Varies with device  Varies with device  
3904  Communication  August 3, 2018  Varies with device  Varies with device

**15. Find Average Rating of Free and Paid Apps**

In [27]:
group_type=google.groupby(by="Type").mean()
group_type

Rating    Reviews
Type                   
Free   4.188 478643.301
Paid   4.267  11673.312

**16. Display Top  5 Apps Having Maximum Installs**

In [28]:
google.Installs.head()

0        10,000+
1       500,000+
2     5,000,000+
3    50,000,000+
4       100,000+
Name: Installs, dtype: object

In [29]:
installs=[]
for i in google.Installs:
    #print(i.split("+"))
    installs.append(i.split("+")[0])


In [30]:
Installs=[]
for i in installs:
    #print(i.split(","))
    Installs.append(("".join(i.split(","))))

In [31]:
google["Installs"]=Installs

In [32]:
for i in google.Installs:
    if i =="Free":
        google.Installs.replace(i, 0, inplace=True)

In [33]:
google.Installs=google.Installs.astype("int64")

In [34]:
google.sort_values(by="Installs", ascending=False).head()

App       Category  Rating  \
3896                            Subway Surfers           GAME   4.500   
3943                                  Facebook         SOCIAL   4.100   
335   Messenger – Text and Video Chat for Free  COMMUNICATION   4.000   
3523                              Google Drive   PRODUCTIVITY   4.400   
3565                              Google Drive   PRODUCTIVITY   4.400   

          Reviews                Size    Installs  Type Price Content Rating  \
3896 27711703.000                 76M  1000000000  Free     0   Everyone 10+   
3943 78128208.000  Varies with device  1000000000  Free     0           Teen   
335  56642847.000  Varies with device  1000000000  Free     0       Everyone   
3523  2731211.000  Varies with device  1000000000  Free     0       Everyone   
3565  2731211.000  Varies with device  1000000000  Free     0       Everyone   

             Genres    Last Updated         Current Ver         Android Ver  
3896         Arcade   July 12, 2018              1.90.0          4.1 and up  
3943         Social  August 3, 2018  Varies with device  Varies with device  
335   Communication  August 1, 2018  Varies with device  Varies with device  
3523   Productivity  August 6, 2018  Varies with device  Varies with device  
3565   Productivity  August 6, 2018  Varies with device  Varies with device